protocol/yggdrasil.proto

```diff
@@ -68,6 +71,16 @@ message Data {
     string directive = 5;
 }
 
+message APIResponse {
```

Comment:
Comments to be added here, still waiting for the first round of reviews.

Reply:
The API looks good to me. I suggest to get Link's review on this, since he'll have to approve the API when it is contributed back to main branch.